vms/platformvm/vm.go

```diff
@@ -487,9 +487,6 @@ func (vm *VM) Disconnected(ctx context.Context, nodeID ids.NodeID) error {
 	if err := vm.uptimeManager.Disconnect(nodeID); err != nil {
 		return err
 	}
-	if err := vm.state.Commit(); err != nil {
-		return err
-	}
```

Comment:
if we want to we can move this to `SetPreference` (or some periodic go routine like mempool pruning etc). I'd imagine block acceptance should be sufficient enough though.

Reply:
yea good idea we could do that instead. but I slightly think that state should be commited with respect to chain progress.